go/services/reports/pkg/openai/client.go

```diff
@@ -293,17 +295,52 @@ type BatchRequestCount struct {
 }
 
 type Batch struct {
-	Id               string            `json:"id"`
-	InputFileId      string            `json:"input_field_id"`
-	CompletionWindow string            `json:"completion_window"`
-	Status           string            `json:"status"`
-	OutputFileId     string            `json:"output_file_id"`
-	CreatedAt        int64             `json:"created_at"`
-	ExpiresAt        int64             `json:"expires_at"`
-	CompletedAt      int64             `json:"completed_at"`
-	FailedAt         int64             `json:"failed_at"`
-	ExpiredAt        int64             `json:"expired_at"`
-	RequestCounts    BatchRequestCount `json:"request_counts"`
+	Id               string             `json:"id" redis:"id"`
+	InputFileId      string             `json:"input_field_id" redis:"input_field_id"`
+	CompletionWindow string             `json:"completion_window" redis:"completion_window"`
+	Status           string             `json:"status" redis:"status"`
+	Errors           BatchErrorsWrapper `json:"-" redis:"-"`
+	OutputFileId     string             `json:"output_file_id" redis:"output_file_id"`
+	CreatedAt        int64              `json:"created_at" redis:"created_at"`
+	ExpiresAt        int64              `json:"expires_at" redis:"expires_at"`
+	CompletedAt      int64              `json:"completed_at" redis:"completed_at"`
+	FailedAt         int64              `json:"failed_at" redis:"failed_at"`
+	ExpiredAt        int64              `json:"expired_at" redis:"expired_at"`
+	RequestCounts    BatchRequestCount  `json:"-" redis:"-"`
+}
+
+type BatchErrorsWrapper struct {
+	Object string       `json:"object" redis:"object"`
+	Data   []BatchError `json:"data" redis:"data"`
+}
+
+type BatchError struct {
+	Code    string `json:"code" redis:"code"`
+	Message string `json:"message" redis:"message"`
+}
+
+func (b *Batch) isCompleted() bool {
+	if b == nil {
```

Comment:
if b == nil can you call isCompleted() on it?

Reply:
No you can't. Thanks for pointing this out.